### PR TITLE
Added support for deleting individual documents by id

### PR DESCRIPTION
**Pull request by agearhart**
Please let me know if you have any concerns with the code.

### CODE REVIEW

File: elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala [hunk 1]
```diff
@@ -117,6 +117,45 @@ class RestlasticSearchClientTest extends WordSpec with Matchers with ScalaFuture
       }
     }
 
+    "Be able to delete a document that exists" in {
+      indexDocs(Seq(Document("doc1", Map("text" -> "here"))))
+      val resFut = restClient.query(index, tpe, QueryRoot(TermQuery("text", "here"),
+        sortOpt = Some(Seq(("_timestamp", "desc"))), timeout = Some(10)))
+
+      var foundDoc: Option[ElasticJsonDocument] = None
+      whenReady(resFut){res =>
```
Comment (rcoh) [inline elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala]: when ready returns the result of the function. You can do:
```
val foundDoc: ElasticJsonDocument = whenReady(resFut) { res =>
  res.raw.....
}
```

This also simplifies the subsequent code.

File: elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala [hunk 2]
```diff
@@ -117,6 +117,45 @@ class RestlasticSearchClientTest extends WordSpec with Matchers with ScalaFuture
...
+      } else {
+        fail("indexed document not found")
+      }
+    }
+
+    "Not delete a document that does not exist" in {
+      val delFut = restClient.deleteById(index, tpe, "fakeId")
+
+      whenReady(delFut.failed) { e =>
+        e shouldBe a [ElasticErrorResponse]
```
Comment (rcoh) [inline elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala]:
```
e match {
  case e: ElasticErrorResponse => ...
  case other => fail(s"Error expected, got ${other}")
}

File: elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala (hunk 1 above)
Comment (rcoh) [reply elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala]: Another option, ```resFut.futureValue.rawSearch......```

**Review by rcoh — CHANGES_REQUESTED** [review]
Couple of changes to the tests

**Review by agearhart — COMMENTED** [review]
Please let me know if I misunderstood the alteration suggestions.  Thank you for the help!

**Review by rcoh — APPROVED** [review]

**Comment by CCheSumo** [comment]
@pnzrdrgoon Let me know if you need a new branch cut that includes the recent PRs for delete/bulkDelete with document id(s)?

**Comment by agearhart** [comment]
That would be fantastic if possible.  I've built a jar locally for use, but
new version cut would be great!

**Comment by CCheSumo** [comment]
@pnzrdrgoon release cut 1.0.30 contains all your changes.

**Pull request merged**